tests/com/tony/server/AuthenticatorTest.java

```diff
@@ -0,0 +1,74 @@
+package com.tony.server;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class AuthenticatorTest {
+    @Test
+    public void testAddToAuthenticationList() throws Exception {
+        Authenticator authenticator = new Authenticator();
```

Comment:
You should pull creating a new `Authenticator` into a [Before](http://junit.sourceforge.net/javadoc/org/junit/Before.html), since you are doing this in nearly every test.

You might also be able to pull out `authenticator.addToAuthenticationList(new Request("GET", "/blah"));`, but this would depend if having the AuthenticationList stored in the `authenticator` would hurt any other tests that don't care about it.